src/js/data-builder.js

```diff
@@ -188,9 +196,10 @@ const buildObjectNodeRelationship = (renderedNodes, filteredJson) => {
   let relatedData = [];
   filteredJson.forEach((element) => {
     if (element.GRANTED_TO_TYPE === 'ROLE' || element.GRANTED_TO_TYPE === 'USER') {
```

Comment:
explain all of this with commenting

Reply:
This file has a lot of business logic and needs a detailed wiki like documentation to explain the concepts of USER, ROLE, OBJECT, and their relationships. Or extensive code commenting and that need to be its own story as tech debt.

Please add a new story or as part of refactoring effort.